### PR TITLE
Send replies

### PR DESCRIPTION
**Pull request by the-Bruce**
Add the ability to the library to send replies. Demonstrated in reminders

### CODE REVIEW

**Review by L0neGamer — APPROVED** [review]
Tested, everything seems to work as expected

File: src/Tablebot/Plugins/Reminder.hs [hunk 1]
```diff
@@ -124,7 +124,7 @@ reminderCron = do
             Left _ -> pure ()
             Right mess -> do
               let (Snowflake uid) = userId (messageAuthor mess)
-              sendMessage mess $
+              sendCustomReplyMessage mess (Snowflake mid) False $
```
Comment (finnbar) [inline src/Tablebot/Plugins/Reminder.hs:127]:
```suggestion
              sendCustomReplyMessage mess (Snowflake mid) True $
```
from discussion in Discord

File: src/Tablebot/Plugin/Discord.hs [hunk 2]
```diff
@@ -59,6 +61,37 @@ sendMessage m t = do
...
+  Bool ->
+  Text ->
+  EnvDatabaseDiscord s ()
+sendCustomReplyMessage m mid fail' t = do
+  let ref = MessageReference (Just mid) Nothing Nothing fail'
+  res <- liftDiscord . restCall $ R.CreateMessageDetailed (messageChannel m) (R.MessageDetailedOpts t False Nothing Nothing Nothing (Just ref))
+  case res of
+    Left _ -> throw $ MessageSendException "Failed to send message."
+    Right _ -> return ()
+
```
Comment (finnbar) [inline src/Tablebot/Plugin/Discord.hs:94]: This all looks strong and works from brief testing (it's the reminder changes that are causing issues)

File: src/Tablebot/Plugins/Reminder.hs [hunk 3]
```diff
@@ -124,7 +124,7 @@ reminderCron = do
             Left _ -> pure ()
```
Comment (finnbar) [inline src/Tablebot/Plugins/Reminder.hs:124]: This isn't caused by your PR (it was a bug from before) - but if a reminder command's original message (the starting `.remind ...` message) is deleted, the reminder will never be reminded about and will be brought up by the database with every run of the cron job without actually doing the reminder.

**Review by finnbar — CHANGES_REQUESTED** [review]
A bug arose from this, for which a solution is provided. There is also an older bug that may as well be fixed with these reminder improvements.

Also, as a quick improvement, could you provide an admin-only `.remind delete` command or similar?

**Pull request merged**